src/main/webapp/filter_entry.js

```diff
@@ -0,0 +1,105 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/* 
+ * FilterEntry displays an area for the user to first specify the field they want to filter by,
+ * and then the keyword they want to apply to that filter field
+ */
+
+class FilterEntry{
+ 
+  // Constructs a new filter entry box
+  constructor(parentFilterArea) { 
+  
+    this.activeFilterArea = parentFilterArea;
+
+    /* filterEntryArea is displayed when the user clicks add filter. It first asks for a type of filter, 
+       then asks the user to enter the keyword for that filter type */
+    this.filterEntryArea = document.createElement("div");
+    this.filterEntryArea.setAttribute("class", "filter-tag-area");
+    this.filterEntryArea.setAttribute("id", "filter-entry");
+
+    /* The user selects the type of property they want to filter by in filterFieldInput*/
+    this.filterFieldInput = document.createElement("input");
+    this.filterFieldInput.setAttribute("list", "filter-datalist");
+    this.filterFieldInput.setAttribute("class", "filter-input-area");
+    this.filterFieldInput.setAttribute("placeholder", "Filter by:");
+    this.filterFieldInput.addEventListener('keypress', (e) => {
+      if (e.key === 'Enter') {
```

Comment:
nit : i find the following structure more readable (here and elsewhere)

(e) -> {
  if (e.key !== 'Enter') {
    return;
  }
  // e.key was enter, do the rest
}


This makes it so I (the reader) don't have to remember the scope of the code inside the curly brace. In other words, as written I would read this code as:

if the key is enter then i do the following, x, y, and z.

In the change I proposed, you would read it as

if the key is not enter, there is nothing to do, return
do x, y, z

If you think of code as branches off a tree what you are trying to do is keep the branch factor uncomplicated and have as many things in the "trunk" as possible

In this case, it's not as important, but it does become much more important if the stuff in side the if gets more complicated (has other branches).